src/h5web/vis-packs/core/utils.ts

```diff
@@ -365,3 +365,30 @@ export function getSliceSelection(
   // Create slice selection string from dim mapping - e.g. [0, 'y', 'x'] => "0,:,:"
   return dimMapping.map((dim) => (isAxis(dim) ? ':' : dim)).join(',');
 }
+
+export function computeSameRatioDomain(
+  availableSize: Size,
+  abscissaDomain: Domain,
+  ordinateDomain: Domain
+): [Domain, Domain] {
+  const { width, height } = availableSize;
+
+  const abscissaRatio = width / (abscissaDomain[1] - abscissaDomain[0]);
+  const ordinateRatio = height / (ordinateDomain[1] - ordinateDomain[0]);
+
+  if (abscissaRatio > ordinateRatio) {
+    const x0 =
+      (abscissaDomain[0] + abscissaDomain[1] - width / ordinateRatio) / 2;
+    const x1 = x0 + availableSize.width / ordinateRatio;
+    return [[x0, x1], ordinateDomain];
+  }
+
+  if (abscissaRatio < ordinateRatio) {
+    const y0 =
+      (ordinateDomain[0] + ordinateDomain[1] - height / abscissaRatio) / 2;
+    const y1 = y0 + height / abscissaRatio;
+    return [abscissaDomain, [y0, y1]];
+  }
+
+  return [abscissaDomain, ordinateDomain];
```

Comment:
If both ratios are already equal, all is well and we simply return the domains :shrug: